File: Code/Constants/Faction.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace watchtower.Constants {

    public sealed class Faction {

        public const short UNKNOWN = -1;

        public const short VS = 1;

        public const short NC = 2;

        public const short TR = 3;

        public const short NS = 4;

        public readonly static List<short> All = new() { VS, NC, TR, NS };

        public static string GetName(short factionID) {
            switch (factionID) {
                case UNKNOWN: return "unknown";
                case VS: return "VS";
                case NC: return "NC";
                case TR: return "TR";
                case NS: return "NS";
                default: return $"Unchecked {factionID}";
            }
        }

    }
}

File: src/util/Faction.ts

export default class FactionUtils {

    public static VS: number = 1;

    public static NC: number = 2;

    public static TR: number = 3;

    public static NS: number = 4;

    public static getName(factionID: number): string {
        if (factionID == FactionUtils.VS) {
            return "VS";
        } else if (factionID == FactionUtils.NC) {
            return "NC";
        } else if (factionID == FactionUtils.TR) {
            return "TR";
        } else if (factionID == FactionUtils.NS) {
            return "NS";
        } else {
            return `Unchecked ${factionID}`;
        }
    }

}